content/blog/2018/07/2018-07-12-simple-pull-request-plugin.adoc

```diff
@@ -0,0 +1,190 @@
+---
+layout: post
+title: "Pipeline as YAML: Alpha release"
+tags:
+- gsoc2018
+- plugin
+- pipeline
+- yaml
+author: abhishek_gautam
+---
+
+=== About me
+
+I am Abhishek Gautam, 3rd year student from Visvesvaraya National Institute of
+technology, India, Nagpur. I was a member of ACM Chapter and Google student developer club of my
+college. I am passionate about automation.
+
+=== Mentors
+
+* link:https://github.com/oleg-nenashev[Oleg Nenashev] (Org Admin)
+* link:https://github.com/martinda[Martin d'Anjou]
+* link:https://github.com/kwhetstone[Kristin Whetstone]
+* link:https://github.com/grandvizier[Jeff Knurek]
+
+=== Project Summary
+
+This is a GSoC 2018 project.
+
+This project aims to develop a pull request Job Plugin. Users should be able to
+configure job type using YAML file placed in root directory of the
+Git repository being the subject of the pull request. The plugin should interact with various
+platforms like Bitbucket, Github, Gitlab, etc whenever a pull request is created or updated.
+
+Plugin detects the presence of certain types of the report at conventional locations,
+and publish them automatically. If the reports are not present at conventional location,
+can specify the location using the YAML file.
+
+==== Benefits to the community
+
+* Project administrators will be able to handle builds for pull requests more easily.
+* Build specifications for pull request can be written in a concise declarative format.
+* Build reports will be automatically published to Github, Bitbucket, etc.
+* Build status updates will be sent to git servers automatically.
+* Users will not have to deal with pipeline code.
+* If there will be no merge conflicts or build failures, the PR can be merged into target branch.
+
+=== Phase 1 blog post
+
+For information of
+
+. Design of the plugin
+. What steps this plugin follow
+. What was done in codeing Phase 1
+. How to run the plugin.
+
+Please see link:https://jenkins.io/blog/2018/06/15/simple-pull-request-plugin/[Phase 1 blog post]
+
+=== Implementations till now
+
+Alpha version of the plugin is released. It supports all features of Multi-Branch Pipeline and offers the following features.
+
+Build description is defined via YAML file stored within the SCM repo. This plugin
+will depend on GitHub plugin, Bitbucket plugin, Gitlab plugin if users will be
+using respective paltfroms for their repositories.
```

Comment:
https://www.jetbrains.com/help/idea/using-code-editor.html#spellchecking

Reply:
Actually intellij have spell checking. I used Atom for this.

Reply:
Ok, try this
https://discuss.atom.io/t/how-to-make-spell-check-word/41235/4 for the `adoc` file type

Reply:
Thanks :+1: